manager/dispatcher/dispatcher.go

```diff
@@ -216,6 +217,9 @@ func (d *Dispatcher) Run(ctx context.Context) error {
 
 	defer cancel()
 	d.ctx, d.cancel = context.WithCancel(ctx)
+	ctx = d.ctx
```

Comment:
It is weird, but it's been like this for awhile.

It's not super easy to avoid storing a context, since a lot of methods wait for this context to be cancelled. We migrated away from a stored context in `raft` by using a `WithContext` function that uses a goroutine to cancel a context when a certain channel is closed. We could potentially do something similar here, but it should be a different PR IMHO.